07_errors/jimbotech/error_test.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+func (s *storesSuite) TestReadFailure() {
+	pup2, err := s.store.ReadPuppy(1)
+	s.Require().Nil(pup2)
+	s.Require().Equal(ErrIDNotFound, err)
+	s.Require().NotEmpty(err.Error())
+}
+
+func (s *storesSuite) TestNegReadFailure() {
+	pup2, err := s.store.ReadPuppy(-1)
+	s.Require().Nil(pup2)
+	s.Require().Equal(ErrValueBelowZero, err)
+}
+
+func (s *storesSuite) TestUpdateError() {
+	create(s)
+	pup2 := Puppy{Breed: "kelpie", Colour: "black", Value: "indispensable"}
+	err := s.store.UpdatePuppy(1, &pup2)
+	s.Require().Equal(ErrIDNotFound, err)
+}
+
+func (s *storesSuite) TestNegUpdateError() {
+	create(s)
+	pup2 := Puppy{Breed: "kelpie", Colour: "black", Value: "indispensable"}
+	err := s.store.UpdatePuppy(-1, &pup2)
+	s.Require().Equal(ErrValueBelowZero, err)
+}
+
+func (s *storesSuite) TestNegDeleteFailure() {
+	err := s.store.DeletePuppy(-1)
+	s.Require().Equal(ErrValueBelowZero, err)
+}
+
+func (s *storesSuite) TestDeleteFailure() {
+	err := s.store.DeletePuppy(1)
+	s.Require().Equal(ErrIDNotFound, err)
+}
+
+func (s *storesSuite) TestError() {
```

Comment:
Should be called `TestErrorNil` or something.

Reply:
Renamed